requester/requester.go

```diff
@@ -82,35 +83,37 @@ type Work struct {
 	// Optional.
 	ProxyAddr *url.URL
 
-	//progressChan is used to update the progress counter
-	progressChan chan int
+	//progressCounter keeps count of completed requests for progress bar
+	progressCounter int32
```

Comment:
Updating counter atomically instead of channel to avoid buffer full situation and changed to single channel stop instead of two channels